src/gsy_e/gsy_e_core/non_p2p_handler.py

```diff
@@ -1,19 +1,35 @@
 from gsy_framework.constants_limits import GlobalConfig
+from gsy_framework.exceptions import GSyException
 
 
 class NonP2PHandler:
     """Handles non-p2p case"""
 
     def __init__(self, scenario: dict):
         self.non_p2p_scenario = scenario
+        self._energy_sell_rate = 0.0
+        self._energy_buy_rate = 0.0
+        self._get_energy_rates_from_infinite_bus(scenario)
```

Comment:
For now this is good. However, keep in mind that in the future we might need to set different energy rates per home. We will deal with it when we need it though, nothing to do for now.

Reply:
Yes, having different rates per home will lead to more changes anyway. Let's deal with this then. Thanks for pointing out though.